nginx/ngx_http_js_module.c

```diff
@@ -4678,8 +4683,19 @@ ngx_http_js_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         }
     }
 
+    if (cf->args->nelts == 4) {
+        if (ngx_strncmp(value[3].data, "nocache", value[3].len) == 0) {
```

Comment:
check here the exact length: 7.

```
if (ngx_strcmp(value[3].data, "nocache") == 0) {
```

Otherwise partial matches will work too. The same with stream module.

Reply:
Thanks, I was worried about out of bounds reading but other parts of nginx's source code show this is the right way 👍